src/utils/ecuUtils.ts

```
@@ -2,6 +2,56 @@
+    log.error(
+      `[ecuUtils] ${fnName} received invalid input type: ${typeof input}`,
+    );
+  }
+};
+
+const flatten = (arr: unknown): number[] => {
+  if (!Array.isArray(arr)) return [];
+
+  return arr.reduce<number[]>((acc, item) => {
+    if (Array.isArray(item)) {
+      return [...acc, ...flatten(item)];
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)